smrt/inputs/make_bedrock.py

```diff
@@ -0,0 +1,236 @@
+    """
+    Returns the relative complex permittivity for various subglacial materials,
+    based on values from Table 1 of Christianson et al. (2016) http://dx.doi.org/10.1002/2015JF003806 . 
+    
+    This function uses a dictionary (CHRISTIANSON_2016_DATA) to look up values.
+    Data is for frequency of 500MHZ, no prescribed temperature.
+    """
+    # Angular frequency
+    angular_frequency = 2 * np.pi * 5E8 #500MHZ measurment
+
+    # Pre-calculate the constant factor for converting conductivity to relative imaginary permittivity
+    conductivity_to_imaginary_factor = 1 / (angular_frequency * PERMITTIVITY_OF_FREE_SPACE)
+
+    try:
+        real_permittivity, conductivity = CHRISTIANSON_2016_DATA[material_type]
+    except KeyError as e:
+        valid_types = list(CHRISTIANSON_2016_DATA.keys())
+        raise SMRTError(
+            f"The material type '{material_type}' is not recognized or implemented. "
+            f"Please refer to the function's documentation for valid options: {', '.join(valid_types)}."
+        ) from e
+
+    imaginary_permittivity = conductivity * conductivity_to_imaginary_factor
+
+    # Return relative complex permittivity: real part + j * imaginary part
+    return complex(real_permittivity, imaginary_permittivity)
+
+
+# --- 3. BUILD FUNCTION (MAKE_BEDROCK) ---
+
+NAMED_PERMITTIVITY_MODELS: Dict[str, Callable[[Any], complex]] = {
+    'hartlieb': constant_permittivity_Hartlieb_2016,
+    'christianson': constant_permittivity_Christianson_2016,
+}
+
+def make_bedrock(
+    substrate_model: Union[str, type],
+    permittivity_model: Union[str, complex, Number, Callable],
+    bedrock_type: Optional[str] = None,
+    temperature: Optional[float] = None,
+    frequency_Hz: Optional[float] = None,
+    **kwargs: Any
+) -> Substrate:
+    """
+    Constructs a bedrock instance based on a given surface electromagnetic model, a rock type and parameters.
+    
+    :param substrate_model: The name or class of the substrate model (e.g., "flat").
+    :param permittivity_model: The permittivity model, which can be:
+        - A string name (e.g., "hartlieb", "christianson").
+        - A constant number (real or complex).
+        - A callable function (frequency_Hz, temperature) -> permittivity.
+    :param bedrock_type: The specific material type required by named string models (e.g., "granite", "frozen bedrock").
+    :param temperature: The physical temperature of the bedrock (K). Mandatory.
+    :param frequency_Hz: The frequency (Hz) to use when calculating complex permittivity for conductivity-based models. 
+        Note that this parameter is ignored by models with fixed measurement frequencies (e.g., "christianson"). Optional.
+    :param kwargs: Additional keyword arguments passed to the final substrate model constructor.
+    :raises SMRTError: If required parameters are missing or models are unrecognized.
+    :return: An instance of a Substrate model.
+    """
+
+    if temperature is None:
```

Comment:
make this argument non optional and remove this code

Reply:
In the argument function, temperature is option. And in the body, we have:
```python
if temperature is None:
 raise SMRTError("Temperature must be provided for bedrock.")
```
isn't it a contradiction ?